### PR TITLE
Extend the `filter()` documentation slightly

### PR DESCRIPTION
**Pull request by The-Mule**
People tend to try using parentheses in their filter expressions for obvious reasons. But it is not possible. I feel like this should be mentioned in the documentation. In fact, with DNF you never need parentheses to express any expression and I think this fact also needs to be noted there.

### CODE REVIEW

File: fmf/utils.py
Comment (lukaszachy) [inline fmf/utils.py:334]: @The-Mule Could you explain why is this change necessary, this isn't just PR for docs then.

**Review by lukaszachy — CHANGES_REQUESTED** [review]

File: fmf/utils.py
Comment (The-Mule) [reply fmf/utils.py:334]: Uh oh, good catch, this slipped in by mistake, sorry - this was bit related to https://github.com/teemtee/fmf/pull/219#issuecomment-2091219335, I'll remove it.

Comment (The-Mule) [reply fmf/utils.py]: Fixed.

File: fmf/utils.py
Comment (psss) [inline fmf/utils.py:217]:
```suggestion
    Filter supports disjunctive normal form (DNF) with '|' used for OR,
```

Let's put the abbreviation after the full term?

File: fmf/utils.py
Comment (psss) [inline fmf/utils.py:225]: Parenthesis are supported in keys or values? Do you mean like regular characters? 'cause, I think currently there's not support for grouping at all.

File: fmf/utils.py
Comment (psss) [inline fmf/utils.py:228]:
```suggestion
    to express '(tag: A | tag: B) & tag: C' use the following filter::
```

Double colons are need for the online documentation to be properly rendered:
* https://fmf--235.org.readthedocs.build/en/235/modules.html#fmf.filter

File: fmf/utils.py
Comment (psss) [inline fmf/utils.py:248]: So I guess, the `|` could/should be covered by #220 which should be included in the `1.4` release as well. Shall we extend the pull request to support `&` escaping as well?

**Review by psss — CHANGES_REQUESTED** [review]
Thanks for improving the docs. Added a couple of questions and suggestions.

**Comment by lukaszachy** [comment]
Depends on #220, we have to adjust docs once that is merged.

**Comment by The-Mule** [comment]
Thanks for the fix, Petre, let me also rebase it now.

**Comment by psss** [comment]
Thanks for the rebase, going to squash the changes now.

**Review by lukaszachy — APPROVED** [review]
One nitpick, but no point in blocking on that

**Pull request merged**
